### ISSUE TITLE
np.timedelta64 should not be an subclass of np.integer

### ISSUE THREAD

**Issue by shoyer** [numpy/numpy#10685]
This is particularly inconsistent because this isn't the case for datetime64:
```
In [12]: issubclass(np.timedelta64, np.integer)
Out[12]: True

In [13]: issubclass(np.datetime64, np.integer)
Out[13]: False
```

Looking at the subclass hierarchy, we observe:
```
In [15]: np.datetime64.mro()
Out[15]: [numpy.datetime64, numpy.generic, object]

In [16]: np.timedelta64.mro()
Out[16]:
[numpy.timedelta64,
 numpy.signedinteger,
 numpy.integer,
 numpy.number,
 numpy.generic,
 object]
```

I would suggest that we make `np.timedelta64` also inherit directly from `np.generic`.

I'm sure there was a reason for this originally, but in my experience this sort of sloppiness around type casting leads to bugs and confusion.

**Comment by mhvk** [comment]
Agreed, `timedelta64` effectively has a unit, so cannot be a subclass of integer. The only thing that might make sense is to have both `datetime64` and `timedelta64` inherit from a common base class.

**Comment by shoyer** [comment]
I put up an attempt to fix this by adding a base class in #10701.

**Comment by eric-wieser** [comment]
> Agreed, timedelta64 effectively has a unit, so cannot be a subclass of integer.

I'm not sure I agree with this. I think there's value to indicating "this is a quantized numeric type", and `np.integer` is the best way we have to do this.

I do think that we shouldn't allow `int + timedelta64` in future

**Comment by mhvk** [comment]
That sounds like subclassing from `PyTimeIntegerArrType_Type` may not be so crazy -- @eric-wieser: maybe see what you think about the discussion in #10701.

**Comment by njsmith** [comment]
> I think there's value to indicating "this is a quantized numeric type"

I guess the argument would be that it's quantized, but not numeric, because numbers are abstract and dimensionless.

**Comment by mattip** [comment]
In a triage we agreed this is desirable. Someone might be relying on this, one way to find out is to move forward with gh-10701 and wait for the reports of breakage. It is not clear if we could add a deprecation warning somehow.
